kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/async/ChildRecord.java

```diff
@@ -0,0 +1,20 @@
+package org.hypertrace.core.kafkastreams.framework.async;
+
+import javax.annotation.Nullable;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.kafka.streams.processor.api.Record;
+
+/**
+ * Encapsulates the {@link Record record} and the {@link String childName} used by the {@link
+ * AsyncProcessor} to forward messages to the downstream processor
+ *
+ * @param <K> type of the record key
+ * @param <V> type of the record value
+ */
+@AllArgsConstructor
+@Getter
+public class ChildRecord<K, V> {
```

Comment:
Any better name? This is exposed to clients so better start with a good name than refactoring later.

Don't have a better suggestion :(